### PR TITLE
Custom MP #1: Create `measurements/` folder

### PR DESCRIPTION
**Pull request by AlbertMitjans**
1. Create `measurements/` folder.
2. Move `measurements.py` file inside it.
3. Separate measurements into different files.

### CODE REVIEW

File: pennylane/measurements/measurements.py [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+
+    @property
+    def control_value(self):
+        """The control value to consider for the measurement outcome."""
+        return self._control_value
+
+    @property
+    def measurements(self):
+        """List of all measurements this MeasurementValue depends on."""
+        return [self._depends_on]
```
Comment (albi3ro) [inline pennylane/measurements/measurements.py:739]: This would belong in `mid_measure.py`.

Comment (AlbertMitjans) [reply pennylane/measurements/measurements.py]: Moved!

File: tests/measurements/test_measurements.py [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+def test_shape_unrecognized_error():
+    """Test that querying the shape of a measurement process with an
+    unrecognized return type raises an error."""
+    mp = MeasurementProcess("NotValidReturnType")
+    with pytest.raises(qml.QuantumFunctionError, match="Cannot deduce the shape"):
+        mp.shape()
+
+
+class TestMeasurementValue:
```
Comment (albi3ro) [inline tests/measurements/test_measurements.py:93]: This class should be moved to `test_mid_measure.py`.

File: tests/measurements/test_probs.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        """Test that an error is raised if a device is not passed when querying
+        the shape of certain measurements."""
+        with pytest.raises(
+            MeasurementShapeError,
+            match="The device argument is required to obtain the shape of the measurement process",
+        ):
+            measurement.shape()
+
+
+class TestBetaProbs:
```
Comment (albi3ro) [inline tests/measurements/test_probs.py:62]: Not exactly "beta" anymore, as this became the behaviour two years ago. Should be fine to merge this with the other class.

File: tests/measurements/test_measurements.py [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+        assert q.queue[0] is tensor_op
+        meas_proc = q.queue[-1]
+        assert isinstance(meas_proc, MeasurementProcess)
+        assert meas_proc.return_type == return_type
+
+        assert q.get_info(tensor_op) == {"owns": (A, B), "owner": meas_proc}
+
+
+@pytest.mark.parametrize("stat_func", [expval, var, sample])
+class TestBetaStatisticsError:
```
Comment (albi3ro) [inline tests/measurements/test_measurements.py:249]: Not exactly "beta" if its been the behaviour for two years.  I think this test should just merged into another test class.

**Review by albi3ro — COMMENTED** [review]
Since we are just moving things around, I don't think we need a changelog entry.

My one blocking comment is moving `TestMeasurementValue` into the `test_mid_measure.py` file.

At a later point we should add more unittests, but we can do that during the upcoming changes.

File: tests/measurements/test_measurements.py (hunk 2 above)
Comment (AlbertMitjans) [reply tests/measurements/test_measurements.py]: Done!

File: tests/measurements/test_probs.py (hunk 3 above)
Comment (AlbertMitjans) [reply tests/measurements/test_probs.py]: Done!

File: tests/measurements/test_measurements.py (hunk 4 above)
Comment (AlbertMitjans) [reply tests/measurements/test_measurements.py]: Done!

**Review by albi3ro — APPROVED** [review]
Looks good to me :)

**Pull request merged**